src/snowflake/snowpark/modin/plugin/_internal/indexing_utils.py

```diff
@@ -911,21 +912,25 @@ def get_valid_col_positions_from_col_labels(
         indexer = []
         for col_loc_item in col_loc:
             # for each locator, we perform prefix matching for multiindex
-            try:
-                if is_scalar(col_loc_item):
-                    col_loc_item = (col_loc_item,)
-                item_indexer = columns.get_locs(col_loc_item)
-            except KeyError:
-                # Note Snowpark pandas will ignore out-of-bound labels
-                item_indexer = []
+            if is_scalar(col_loc_item):
+                col_loc_item = (col_loc_item,)
+            # May throw KeyError if any labels are not found in columns
+            item_indexer = columns.get_locs(col_loc_item)
             indexer.extend(item_indexer)
     else:
         if is_scalar(col_loc):
+            if col_loc not in columns:
+                raise KeyError(f"{col_loc}")
             col_loc = [col_loc]
         elif isinstance(col_loc, tuple):
             col_loc = [col_loc] if col_loc in columns else list(col_loc)
-
-        # Note Snowpark pandas will ignore out-of-bound labels
+        # Throw a KeyError in case there are any missing column labels
```

Comment:
nit: we probably can refactor LL 922 and LL 928-933 into a helper function `check_columns(columns, col_loc)` to perform the check. Might make the logic easier to read here and we prob. can reuse the code later somewhere else.

Reply:
The logic for handling scalar values and list values are different, and they throw different messages, so I don't see a ton of value in refactoring the `if label in columns` logic since it's pretty readable as is.